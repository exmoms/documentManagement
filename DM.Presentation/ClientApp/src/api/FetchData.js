/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
