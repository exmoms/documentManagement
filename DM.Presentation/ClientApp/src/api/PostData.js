/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
