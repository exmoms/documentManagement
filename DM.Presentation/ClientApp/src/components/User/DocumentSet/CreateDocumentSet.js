/**
 * @file Contains CreateDocumentSet component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Fragment } from "react";
import { Redirect } from "react-router";
import { TextField } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import { postDataToAPI } from "../../../api/PostData";
import { withTranslation } from "react-i18next";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogActions from "@material-ui/core/DialogActions";

const styles = (theme) => ({
  root: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  box: {
    padding: "10px",
  },
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: 300,
  },
  Button: {
    marginTop: "70px",
  },
});

class CreateDocumentSet extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      redirectToDocumentSet: false,
      openNotifiction: false,
      error: false,
      errorMessage: [],
    };

    this.handleClose = this.props.action;
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidUpdate() {
    if (this.state.redirectToDocumentSet) {
      this.setState(
        {
          redirectToDocumentSet: false,
        },
        this.handleClose()
      );
    }
  }

  handleChange(event) {
    this.setState({ name: event.target.value });
  }

  handleSubmit = async (event) => {
    let set = this.props.set;
    let response = {};
    if (set.id === 0) {
      response = await postDataToAPI("/api/DocumentSet/AddNewDocumentSet", {
        Name: this.state.name,
      });
    } else {
      response = await postDataToAPI("/api/DocumentSet/AddNewDocumentSet", {
        Name: this.state.name,
        ParentDocumentSet: set.id,
      });
    }
    if (response.ok) {
      this.setState(
        {
          redirectToDocumentSet: true,
          name: "",
          openNotifiction: true,
        } //,
        //this.handleClose()
      );
    } else {
      let json = await response.json();
      this.setState(
        {
          redirectToDocumentSet: true,
          name: "",
          openNotifiction: true,
          error: true,
          errorMessage: json.error,
        } //,
        // this.handleClose()
      );
    }
  };

  t = this.props["t"];
  render() {
    const { classes } = this.props;
    let lang = localStorage.getItem("i18nextLng");
    return (
      <Fragment>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.handleClose()}
          maxWidth={"xs"}
        >
          <div
            style={
              lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
            }
          >
            <DialogTitle id="simple-dialog-title">
              {this.t("documentset_addnewdocumentset")}
            </DialogTitle>
          </div>
          <DialogContent>
            <div className={classes.root}>
              <div className={classes.box}>
                <div
                  style={
                    lang === "ar"
                      ? { textAlign: "right" }
                      : { textAlign: "left" }
                  }
                >
                  {this.t("documentset_entername")}
                </div>
                <TextField
                  id="standard-basic"
                  className={classes.textField}
                  required
                  helperText={this.t("hypertext_required")}
                  label={this.t("namelabel")}
                  fullWidth
                  value={this.state.name}
                  onChange={this.handleChange}
                />
                <br />
                <Button
                  variant="contained"
                  className={classes.Button}
                  color="primary"
                  fullWidth
                  onClick={this.handleSubmit}
                >
                  {" "}
                  {this.t("button_submit")}
                </Button>
              </div>
            </div>
          </DialogContent>
          <DialogActions>
            <Button
              id="CancelButton"
              onClick={() => this.handleClose()}
              color="primary"
            >
              {this.t("cancel")}
            </Button>
          </DialogActions>
        </Dialog>
        {this.state.redirectToDocumentSet && (
          <Redirect
            to={{
              pathname: "/document-set",
              state: {
                id: this.props.set.id,
                name: this.props.set.name,
                openNotifiction: this.state.openNotifiction,
                error: this.state.error,
                errorMessage: this.state.errorMessage,
              },
            }}
          />
        )}
      </Fragment>
    );
  }
}

export default withStyles(styles, { withTheme: true })(
  withTranslation()(CreateDocumentSet)
);
