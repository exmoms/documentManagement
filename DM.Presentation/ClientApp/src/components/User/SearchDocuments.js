import React, { Component } from "react";

import SearchDocumentByValue from "./SearchDocumentByValue";
import SearchDocumentByFreeText from "./SearchDocumentByFreeText";
import { postDataToAPI } from "../../api/PostData";
import { Notification } from "../Admin/Notifications";

import Radio from "@material-ui/core/Radio";
import RadioGroup from "@material-ui/core/RadioGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import FormControl from "@material-ui/core/FormControl";

import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import Paper from "@material-ui/core/Paper";
import { withTranslation } from "react-i18next";

export class SearchDocuments extends Component {
  constructor() {
    super();
    this.state = {
      value: "free-text",
      search_url: "",
      request_payload: {},
      selectedAttributeNames: [],
      openNotifiction: false,
      error: false,
      errorMessage: [],
    };
    this.composeSearchRequest = this.composeSearchRequest.bind(this);
  }

  composeTableColumns = (action) => {
    const { t } = this.props;
    let columns = [];
    let { selectedAttributeNames } = this.state;
    switch (action) {
      case "browse-document":
      case "update-document":
      case "add-to-documentset":
        if (this.state.value === "free-text") {
          columns.push({
            title: t("documenttable_attributename"),
            field: "attributeName",
          });
          columns.push({
            title: t("documenttable_attributevalue"),
            field: "attributeValue",
          });
        } else {
          selectedAttributeNames.forEach((name, index) => {
            columns.push({ title: name, field: name });
          });
        }

        columns.push({
          title: t("documenttable_docid"),
          field: "documentId",
          hidden: true,
        });
        columns.push({
          title: t("documenttable_docname"),
          field: "documentName",
          hidden: true,
        });
        columns.push({
          title: t("documenttable_latestversion"),
          field: "latestVersion",
        });

        break;
      case "show-history":
        columns = [
          { title: t("documenttable_docname"), field: "documentName" },
          {
            title: t("documenttable_docid"),
            field: "documentId",
            hidden: true,
          },
          { title: t("version_message"), field: "versionMessage" },
          { title: t("added_date"), field: "addedDate" },
          {
            title: t("version_id"),
            field: "latestVersion",
          },
        ];
        break;
      default:
    }
    return columns;
  };

  composeTableData = (data) => {
    let list = [];
    if (this.state.value === "free-text") {
      data.forEach((element, index) => {
        element.values.forEach((value) => {
          list.push({
            attributeName: value.metaDataAttributeName,
            attributeValue: value.value,
            documentId: element.documentId,
            documentName: element.documentName,
            latestVersion: element.latestVersion,
          });
        });
      });
    } else {
      data.forEach((element, index) => {
        let row = {
          documentId: element.documentId,
          documentName: element.documentName,
          latestVersion: element.latestVersion,
        };

        element.values.forEach((value) => {
          Object.defineProperty(row, value.metaDataAttributeName, {
            value: value.value,
            writable: true,
            enumerable: true,
            configurable: true,
          });
        });
        list.push(row);
      });
    }
    return list;
  };

  handleSendRequest = async () => {
    let list = [];
    let columns = [];

    let response = await postDataToAPI(
      this.state.search_url,
      this.state.request_payload
    );

    if (response.ok) {
      let data = await response.json();
      list = this.composeTableData(data);
      columns = this.composeTableColumns(this.props.action);
    }

    this.setState({
      search_url: "",
      request_payload: {},
      selectedAttributeNames: [],
    });

    this.props.getDocumentList(list, columns);
  };

  componentDidUpdate() {
    if (this.state.search_url !== "") {
      this.handleSendRequest();
    }
  }

  handleChange = (event) => {
    this.setState(
      { value: event.target.value },
      this.props.getDocumentList([])
    );
  };

  composeSearchRequest = (url, payload, selectedAttributeNames = []) => {
    if (payload !== null) {
      this.setState({
        search_url: url,
        request_payload: payload,
        selectedAttributeNames: selectedAttributeNames,
        openNotifiction: false,
        error: false,
        errorMessage: [],
      });
    } else {
      this.setState({
        search_url: url,
        request_payload: {},
        selectedAttributeNames: selectedAttributeNames,
        openNotifiction: false,
        error: false,
        errorMessage: [],
      });
    }
  };

  handleError = (error) => {
    this.setState({
      openNotifiction: error.openNotifiction,
      error: error.error,
      errorMessage: error.errorMessage,
    });
  };

  showFilter = () => {
    switch (this.state.value) {
      case "free-text":
        return (
          <SearchDocumentByFreeText
            composeSearchRequest={this.composeSearchRequest}
            handleError={this.handleError}
            modelId={this.props.modelId}
            setIdExclude={this.props.setIdExclude}
          />
        );

      case "value":
        return (
          <SearchDocumentByValue
            composeSearchRequest={this.composeSearchRequest}
            handleError={this.handleError}
            modelId={this.props.modelId}
            setIdExclude={this.props.setIdExclude}
          />
        );
      default:
        return "";
    }
  };

  setOpenNotification = (value) => {
    this.setState({ openNotifiction: value });
  };

  render() {
    //const { t } = this.props["t"];
    const { t } = this.props;
    return (
      <div>
        <TableContainer component={Paper}>
          <Table aria-label="simple table">
            <TableHead>
              <TableRow>
                <TableCell>
                  <FormControl component="fieldset" margin="dense">
                    {t("tablecell_searchby")}
                    <RadioGroup
                      aria-label="gender"
                      name="gender1"
                      value={this.state.value}
                      onChange={this.handleChange}
                      row
                    >
                      <FormControlLabel
                        value="free-text"
                        control={<Radio />}
                        label={t("label_freetext")}
                      />
                      <FormControlLabel
                        value="value"
                        control={<Radio />}
                        label={t("label_value")}
                      />
                    </RadioGroup>
                  </FormControl>
                </TableCell>
                {/* <TableCell>{t("tablecell_searchfrom")}</TableCell> */}
              </TableRow>
            </TableHead>
            <TableBody>
              <TableRow key="search-form">
                {/* <TableCell></TableCell> */}
                <TableCell>
                  {this.showFilter()}
                  <Notification
                    open={this.state.openNotifiction} // must be a "state" to show notification message [true or false]
                    setOpen={this.setOpenNotification} // this function changes the state "open" value.
                    error={this.state.error} // if error message put [true] else for success put [false]
                    errorMessage={this.state.errorMessage} // the recived error message (array of string) from server.
                  />
                </TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </TableContainer>
      </div>
    );
  }
}

export default withTranslation()(SearchDocuments);
