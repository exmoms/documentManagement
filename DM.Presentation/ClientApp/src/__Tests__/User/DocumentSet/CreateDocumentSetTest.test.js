import React from "react";
import ReactDom from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import { createMount } from '@material-ui/core/test-utils';
import Dialog from "@material-ui/core/Dialog";
import { Fragment } from "react";
import ListItem from "@material-ui/core/ListItem";
import CreateDocumentSet from "../../../components/User/DocumentSet/CreateDocumentSet";
import SearchDocuments from "../../../components/User/SearchDocuments";
import DocumentTable from "../../../components/User/DocumentTable";
import Button from "@material-ui/core/Button";
import * as FetchMock from "../../../api/FetchData";
import ListItemText from "@material-ui/core/ListItemText";
import * as PostDataMock from "../../../api/PostData";
import List from "@material-ui/core/List";
import { BrowserRouter as Router } from "react-router-dom";
import { Redirect } from "react-router";
import { Typography } from "@material-ui/core";
import DialogContent from "@material-ui/core/DialogContent";
import { TextField } from "@material-ui/core";
afterEach(cleanup);

describe("CreateDocumentSet", () => {
    afterEach(() => {
        jest.clearAllMocks();
      });
    test("renders without crashing", () => {
        let show=true;
        let set={id:1, name:"Name A"};
        let action = jest.fn();
        const div = document.createElement("div");
        ReactDom.render(<CreateDocumentSet show={show} set={set}  action={action}/>, div);
      });

    test("CreateDocumentSet matches snapshot", () => {
        let show=true;
        let set={id:1, name:"Name A"};
        let action = jest.fn();
    const renderedValue = createMount()(
        <CreateDocumentSet show={show}  set={set}  action={action}/>
      );
      expect(renderedValue.html()).toMatchSnapshot();
    });

    test("intializations",  () =>{
        let show=true;
        let set={id:1, name:"Name A"};
        let action = jest.fn();
        let classes ={Typography:"Typography" , root:"root",box:"box",textField:"textField",Button:"Button"}
        let  wrapper = mount(
                <CreateDocumentSet show={show}  set={set} action={action} classes={classes}/>
              );
    expect(wrapper.find(Dialog).at(0).props().open).toBeTruthy();
    expect(wrapper.find(TextField).at(0).props().value).toBe(""); 
    expect(wrapper.find(Redirect).length).toBe(0);
    })

    test("handleChange",  () =>{
        let show=true;
        let set={id:1, name:"Name A"};
        let action = jest.fn();
        let classes ={Typography:"Typography" , root:"root",box:"box",textField:"textField",Button:"Button"}
        let  wrapper = mount(
                <CreateDocumentSet show={show}  set={set} action={action} classes={classes}/>
              );

              act(() => {
                wrapper
                  .find(TextField)
                  .at(0)
                  .props()
                  .onChange({ target: { value: "tested Value" } });
              });
              
            wrapper.update();
            wrapper.render();
    expect(wrapper.find(Dialog).at(0).props().open).toBeTruthy();
    expect(wrapper.find(TextField).at(0).props().value).toBe("tested Value"); 
    expect(wrapper.find(Redirect).length).toBe(0);
    })

    test("handleSubmit functionalities when res is ok",  () =>{
        let show=true;
        let set={id:1, name:"Name A"};
        let action = jest.fn();
          let mockJsonPromisepostDataToAPI={ok:true , json:()=>{ return [] }};
          const mockJsonPromisePost = Promise.resolve(
            mockJsonPromisepostDataToAPI ); 
          const postDataToAPIMock = jest
              .spyOn(PostDataMock, "postDataToAPI").mockImplementation(() => mockJsonPromisePost);
        let wrapper = mount(
            <Router><CreateDocumentSet show={show}  set={set} action={action}/> </Router>);  
            act(() => {
                wrapper
                  .find(TextField)
                  .at(0)
                  .props()
                  .onChange({ target: { value: "tested Value" } });
              });
            wrapper.update();
            wrapper.render();
            return mockJsonPromisePost.then(() => {
             act(() => {
                wrapper
                  .find(Button)
                  .at(0)
                  .props()
                  .onClick({ target: { value: "" } });
              });
            }).then(() => {
            wrapper.update();
            wrapper.render();
            expect(postDataToAPIMock).toHaveBeenCalledTimes(1);
            expect(postDataToAPIMock.mock.calls[0][0]).toBe('/api/DocumentSet/AddNewDocumentSet');
            expect(postDataToAPIMock.mock.calls[0][1]).toStrictEqual({Name:"tested Value" ,"ParentDocumentSet": 1,});
             //redirectToDocumentSet is resetted to false because of update functionality
            expect(wrapper.find(Redirect).length).toBe(0);    
           expect(wrapper.find(TextField).at(0).props().value).toBe("");  
           expect(action).toHaveBeenCalledTimes(1);
          });
    });

    test("handleSubmit functionalities when res isn't ok",  () =>{
        let show=true;
        let set={id:0, name:"Name A"};
        let action = jest.fn();
          let mockJsonPromisepostDataToAPI={ok:false , json:()=>{ return {error:["error"]} }};
          const mockJsonPromisePost = Promise.resolve(
            mockJsonPromisepostDataToAPI ); 
          const postDataToAPIMock = jest
              .spyOn(PostDataMock, "postDataToAPI").mockImplementation(() => mockJsonPromisePost);
        let wrapper = mount(
            <Router><CreateDocumentSet show={show}  set={set} action={action}/> </Router>);  
            act(() => {
                wrapper
                  .find(TextField)
                  .at(0)
                  .props()
                  .onChange({ target: { value: "tested Value" } });
              });
            wrapper.update();
            wrapper.render();
            return mockJsonPromisePost.then(() => {
             act(() => {
                wrapper
                  .find(Button)
                  .at(0)
                  .props()
                  .onClick({ target: { value: "" } });
              });
            }).then(() => {
            wrapper.update();
            wrapper.render();
            expect(postDataToAPIMock).toHaveBeenCalledTimes(1);
            expect(postDataToAPIMock.mock.calls[0][0]).toBe('/api/DocumentSet/AddNewDocumentSet');
            expect(postDataToAPIMock.mock.calls[0][1]).toStrictEqual({Name:"tested Value"});
            //redirectToDocumentSet is resetted to false because of update functionality
            expect(wrapper.find(Redirect).length).toBe(0);   
           expect(wrapper.find(TextField).at(0).props().value).toBe("tested Value");  
           expect(action).toHaveBeenCalledTimes(0);
          });
    });
});