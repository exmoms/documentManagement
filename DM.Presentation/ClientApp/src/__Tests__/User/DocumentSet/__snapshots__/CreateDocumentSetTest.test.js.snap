// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDocumentSet CreateDocumentSet matches snapshot 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" position=\\"centerCenter\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\"><div style=\\"text-align: left;\\"><div class=\\"MuiDialogTitle-root\\" id=\\"simple-dialog-title\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">documentset_addnewdocumentset</h2></div></div><div class=\\"MuiDialogContent-root\\"><div class=\\"withI18nextTranslation(CreateDocumentSet)-root-1\\"><div class=\\"withI18nextTranslation(CreateDocumentSet)-box-2\\"><div style=\\"text-align: left;\\">documentset_entername</div><div class=\\"MuiFormControl-root MuiTextField-root withI18nextTranslation(CreateDocumentSet)-textField-3 MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"standard-basic\\" id=\\"standard-basic-label\\">namelabel<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\">â€‰*</span></label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" aria-describedby=\\"standard-basic-helper-text\\" id=\\"standard-basic\\" required=\\"\\" type=\\"text\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div><p class=\\"MuiFormHelperText-root Mui-required\\" id=\\"standard-basic-helper-text\\">hypertext_required</p></div><br><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained withI18nextTranslation(CreateDocumentSet)-Button-4 MuiButton-containedPrimary MuiButton-fullWidth\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"> button_submit</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary\\" tabindex=\\"0\\" type=\\"button\\" id=\\"CancelButton\\"><span class=\\"MuiButton-label\\">cancel</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
